modules/iot-core/main.tf

```diff
@@ -60,22 +82,22 @@ resource "google_cloudiot_registry" "registry" {
 #---------------------------------------------------------
 
 resource "google_cloudiot_device" "device" {
-  for_each = try(yamldecode(file(var.devices_config.yaml_file)), {})
+  for_each = local.device_config
   name     = each.key
   registry = google_cloudiot_registry.registry.id
 
   credentials {
     public_key {
-      format = var.devices_config.certificate_format
-      key    = file(each.value)
+      format = each.value.certificate_format
```

Comment:
anything that comes from a yaml should use try(), otherwise the module will fail if the yaml doesn't have the field